src/cli/commands/build.ts

```diff
@@ -0,0 +1,173 @@
+    }
+
+    logger.log('Nothing to build.');
+    return;
+  }
+
+  logger.log(`Build configuration:`);
+  logger.log(`- App location: ${chalk.green(appLocation || '')}`);
+  logger.log(`- API location: ${chalk.green(apiLocation || '')}`);
+  logger.log(`- Output location: ${chalk.green(outputLocation || '')}`);
+  logger.log(`- App build command: ${chalk.green(appBuildCommand || '')}`);
+  logger.log(`- API build command: ${chalk.green(apiBuildCommand || '')}`);
+  
+  if (appBuildCommand) {
+    let buildPath = appLocation!;
+    const packageJsonPath = await findUpPackageJsonDir(appLocation!, outputLocation!);
+    if (packageJsonPath) {
+      logger.log(`Found package.json in ${packageJsonPath}`);
+      await installNpmDependencies(packageJsonPath);
+      buildPath = packageJsonPath;
+    }
+
+    logger.log(`Building app with ${chalk.green(appBuildCommand)} in ${chalk.dim(buildPath)}...`);
+    runCommand(appBuildCommand, buildPath!);
+  }
+
+  if (apiBuildCommand) {
+    let buildPath = appLocation!;
+    const packageJsonPath = await findUpPackageJsonDir(appLocation!, apiLocation!);
+    if (packageJsonPath) {
+      logger.log(`Found package.json in ${packageJsonPath}`);
+      await installNpmDependencies(packageJsonPath);
+      buildPath = packageJsonPath;
+    }
+
+    logger.log(`Building api with ${chalk.green(apiBuildCommand)} in ${chalk.dim(buildPath)}...`);
+    runCommand(apiBuildCommand, buildPath!);
+  }
+}
+
+function hasBuildOptionsDefined(options: SWACLIConfig): boolean {
+  if (options.appBuildCommand || options.apiBuildCommand) {
+    return true;
+  }
+  return isUserOrConfigOption('appBuildCommand') || isUserOrConfigOption('apiBuildCommand');
+}
+
+function showAutoErrorMessageAndExit() {
+  logger.error(`Please "swa init" to set your configuration or use options flags to set your`);
+  logger.error(`build commands and paths.`, true);
+}
+
+declare type NpmPackageManager = 'npm' | 'yarn' | 'pnpm';
+
+async function detectPackageManager(basePath: string): Promise<NpmPackageManager> {
+  const hasYarnLock = await pathExists(path.join(basePath, 'yarn.lock'));
+  const hasNpmLock = await pathExists(path.join(basePath, 'package-lock.json'));
+  const hasPnpmLock = await pathExists(path.join(basePath, 'pnpm-lock.yaml'));
+
+  if (hasPnpmLock && !hasNpmLock && !hasYarnLock) {
+    return 'pnpm';
+  }
+
+  if (hasYarnLock && !hasNpmLock) {
+    return 'yarn';
+  }
+
+  return 'npm';
+}
+
+async function installNpmDependencies(packageJsonPath: string): Promise<void> {
+  try {
+    const packageManager = await detectPackageManager(packageJsonPath);
+    logger.log(`Installing dependencies with "${packageManager} install"...`);
+    runCommand(`${packageManager} install`, packageJsonPath);
+  } catch (error) {
+    logger.error(`Cannot install dependencies:`);
+    logger.error(error as Error, true);
+  }
+}
+
+function runCommand(command: string, cwd: string) {
+  execSync(command, {
```

Comment:
we need to make sure the command here is windows safe. We had issues in the past with commands not executing on Windows.

Reply:
I have a few other tools relying on the same code that are working fine on windows for years ;)
We should be able to test it in Cypress anyways